Implement budget UI

junjizhi opened the pull request:
Fixes https://github.com/statechannels/monorepo/issues/1171

The budget comes from mock. Next I will try to wire up the budget update call.

The budget UI looks like this:

![image](https://user-images.githubusercontent.com/2715151/77547177-3bb96b80-6e83-11ea-86d9-52fe642713af.png)

geoknee commented on packages/web3torrent/src/components/torrent-info/download-info/DownloadInfo.tsx:
```diff
@@ -66,11 +68,14 @@ const DownloadInfo: React.FC<DownloadInfoProps> = ({
+              Total Spent: <strong className="total-spent">{prettyPrintWei(totalSpent)}</strong>
+            </p>
+          </section>
+          <ChannelBudget budget={createMockBudget()} />
```
If I am uploading & downloading, it looks like I will see the `<ChannelBudget/>` component twice. Is that what we want?

geoknee replied:
Perhaps it would be best to call the component `<SiteBudget/>` ?

I think it is best thought of as an integral part of the setup of web3torrent, and therefore shown as soon as a budget exists (rather than conditional on what is happening in any particular payment channel, or waiting until torrenting has begun).

junjizhi replied:
> If I am uploading & downloading, it looks like I will see the <ChannelBudget/> component twice. Is that what we want?

Good catch. I thought peer can only be download or upload for the demo, so it would only show once. Anyway, I moved the component up level.

> Perhaps it would be best to call the component <SiteBudget/> ?
I renamed it to `SiteBudgetTable` so that it doesn't have a name conflict with `SiteBudget` type.

geoknee replied:
👍 

(The plan is to allow non seeding peers to upload and download simultaneously)

junjizhi commented:
> Do you think we are displaying redundant information with the Total Spent and Total Received fields (which should be the same as the Spent and Earned fields of your new component)?

I removed the Total Spent / Received in the text and replaced them with the budget table. See the screenshot above.

geoknee commented:
On second thoughts, maybe we are labelling the  information incorrectly here. `inUse.hubAmount` is not the amount that I have received: it is the *maximum* amount I could receive if all of my uploading payment channels were exhausted. Similarly `inUse.playerAmount` is similarly the maximum amount I could pay out if all my downloading channels got exhausted.

geoknee commented:
I'll close this for now but we should reopen when #1302 lands and this component is rebuilt against the new interface.
